lib/mix/tasks/firmware.burn.ex

```diff
@@ -99,7 +110,20 @@ defmodule Mix.Tasks.Firmware.Burn do
   end
 
   defp get_devs do
-    {result, 0} = System.cmd("fwup", ["--detect"])
+    {result, 0} = 
+      if is_wsl?() do
+        path = Path.expand("./") <> "/fwup_devices.txt"
+        path = conv_path_to_windows(path)
+        System.cmd("powershell.exe", ["-Command", "Start-Process powershell.exe -Verb runAs -Wait -ArgumentList \"fwup.exe -D | set-content -encoding UTF8 #{path}\""])
+       
+        devs = File.read("fwup_devices.txt") 
+          |> elem(1)
```

Comment:
I have changed this so instead of immediately accessing the first element it does a pattern match. Hopefully this will make it easier to debug future errors.